pkg/build/apis/build/types.go

```diff
@@ -514,6 +514,10 @@ const (
 	// range of build failures.
 	StatusReasonGenericBuildFailed StatusReason = "GenericBuildFailed"
 
+	// StatusReasonGenericBuildFailed is the reason associated with a broad
+	// range of build failures.
+	StatusReasonOOMKilled StatusReason = "OOMKilled"
```

Comment:
reminder: have to be copied to openshift/api as well

Reply:
Thanks for the reminder, I assume somewhere [here](https://github.com/openshift/api/blob/04a26bf3b8d69c390642c5803fe4cfdb899112aa/build/v1/consts.go)

Btw. it appears to work even without the API change, so just out of pure curiosity, what is the purpose of having the constants duplicated in the API? I didn't find any errors or warnings in the logs.

```bash
$ oc get build
NAME      TYPE      FROM          STATUS                       STARTED          DURATION
bc1-1     Source    Git           Failed (FetchSourceFailed)   13 minutes ago   5s
bc1-2     Source    Git           Failed (FetchSourceFailed)   12 minutes ago   2s
bc1-3     Source    Git@eb6a686   Failed (AssembleFailed)      12 minutes ago   19s
bc1-4     Source    Git           Failed (OOMKilled)           10 minutes ago   2s
bc1-5     Source    Git           Failed (OOMKilled)           5 seconds ago    1s

$ oc describe build bc1-5 | grep 'Status:'
Status:         Failed (Out of memory)
```

Reply:
> what is the purpose of having the constants duplicated in the API? 

putting them in the api repo makes them available to other repos like client tools that want to use them.  Also technically the types in origin are the internal types which do not have any guarantees about not being changed.  Constants defined in the openshift/api repo (the external api) are part of the api spec, meaning we clients can rely on them and we won't change their values.